packages/core/addon/components/navi-visualizations/table.js

```diff
@@ -312,6 +349,156 @@ export default Component.extend({
     return direction;
   },
 
+  /**
+   * Synchronizes the width of visible table headers with real (hidden) table header elements
+   *
+   * @method _syncHeadersWidth
+   * @private
+   * @returns {void}
+   */
+  _syncHeadersWidth() {
+    //in editing mode the real table headers are visible, no need to sync
+    if (get(this, 'isEditingMode')) {
+      return;
+    }
+
+    let widths = Array.from(this.element.querySelectorAll('th')).map(elm => elm.getBoundingClientRect().width);
+
+    let prevWidths = get(this, 'columnsWidth'),
+      widthChanged = widths.length !== prevWidths.length; //if number of columns has changed, set to true to re-set width
+
+    Array.from(this.element.querySelectorAll('div.table-header-row .table-header-cell')).map((headerElm, i) => {
+      let headerElmWidth = parseInt(headerElm.style.width);
+
+      if (!headerElmWidth || widths[i] !== prevWidths[i]) {
+        headerElm.style.width = `${widths[i]}px`;
+        widthChanged = true;
+      }
+    });
+
+    //set total width to headers wrapper div
+    let headerRowElm = this.element.querySelector('div.table-header-row'),
```

Comment:
Could this be cached or does it change too often?

Reply:
So I was wondering if caching selectors is a good practice in Ember like it is in React for example, because in no tutorial so far have I seen this. I'll try here

Reply:
you mean the DOM element itself? it's removed from the DOM on edit mode actually